src/components/ItaliaTheme/View/ViewUtils.jsx

```diff
@@ -13,3 +15,67 @@ export const readingTime = (title, description, htmlBody) => {
 
 export const getHTMLString = (content, locale) =>
   renderToString(<IntlProvider locale={locale}>{content}</IntlProvider>);
+
+//Hook per avere il reading time di una pagina
+export const useReadingTime = (content, documentBody) => {
+  const [readingtime, setReadingtime] = useState(0);
+
+  useEffect(() => {
+    if (documentBody.current) {
+      if (__CLIENT__) {
+        setReadingtime(
+          readingTime(content.title, content.description, documentBody),
+        );
+      }
+    }
+  }, [content.description, content.title, documentBody]);
+
+  return { readingtime, setReadingtime };
+};
+
+//Hook per avere i SideMenu elements di una pagina
+export const useSideMenu = (content, documentBody) => {
+  const [sideMenuElements, setSideMenuElements] = useState(null);
+  const [observer, setObserver] = useState(null);
+  const updateSideMenuOnLoadingBlocks =
+    config.settings?.italiaThemeViewsConfig?.[content['@type']]
+      ?.updateSideMenuOnLoadingBlocks ?? false;
+  const SideMenuComponent =
+    config.settings?.italiaThemeViewsConfig?.[content['@type']]?.sideMenu ??
+    SideMenu;
+
+  useEffect(() => {
+    if (documentBody.current) {
+      if (__CLIENT__) {
+        setSideMenuElements(documentBody.current);
+      }
+    }
+  }, [content.description, content.title, documentBody]);
+
+  //observer is needed for loadable blocks like listing and rss, if you want to use their title's for SideMenu
+  useEffect(() => {
+    const obs = updateSideMenuOnLoadingBlocks
+      ? new MutationObserver((mutationList) => {
+          setSideMenuElements(documentBody.current);
+        })
+      : null;
+    setObserver(obs);
+  }, [setObserver]);
+
+  useEffect(() => {
+    if (!observer && !updateSideMenuOnLoadingBlocks) return;
+
+    observer.observe(documentBody.current, {
+      //attributes: true,
+      childList: true,
+      subtree: true,
+    });
+    return () => {
+      if (observer) {
+        observer.disconnect();
+      }
+    };
+  }, [observer, documentBody, updateSideMenuOnLoadingBlocks]);
```

Comment:
Questi due non si possono fare in un unico useEffect? Evitiamo un po' di rerender dei componenti così, perché altrimenti nel primo rirenderizzi perché setti lo stato, poi fai un altro effetto in cui controlli comunque di avere `updateSideMenuOnLoadingBlocks`. Se nel secondo controlli la config e poi dichiari e usi al volo l'observer dovrebbe andare comunque, o sbaglio?